internal/service/ssoadmin/customer_managed_policy_attachment.go

```diff
@@ -0,0 +1,247 @@
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+const (
+	customerPolicyAttachmentTimeout = 5 * time.Minute
+)
+
+func ResourceCustomerManagedPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCustomerManagedPolicyAttachmentCreate,
+		Read:   resourceCustomerManagedPolicyAttachmentRead,
+		Delete: resourceCustomerManagedPolicyAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"instance_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"permission_set_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"customer_managed_policy_reference": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "/",
+							ForceNew: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCustomerManagedPolicyAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).SSOAdminConn
+
+	instanceArn := d.Get("instance_arn").(string)
+	permissionSetArn := d.Get("permission_set_arn").(string)
+	policyName, policyPath := expandPolicyReference(d.Get("customer_managed_policy_reference").([]interface{}))
+
+	input := &ssoadmin.AttachCustomerManagedPolicyReferenceToPermissionSetInput{
+		InstanceArn:                    aws.String(instanceArn),
+		CustomerManagedPolicyReference: formatPolicyReference(d.Get("customer_managed_policy_reference").([]interface{})),
+		PermissionSetArn:               aws.String(permissionSetArn),
+	}
+
+	err := resource.Retry(customerPolicyAttachmentTimeout, func() *resource.RetryError {
+		var err error
+		_, err = conn.AttachCustomerManagedPolicyReferenceToPermissionSet(input)
+
+		if err != nil {
+			if tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeConflictException) {
+				return resource.RetryableError(err)
+			}
+			if tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeThrottlingException) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+
+	})
+
+	if tfresource.TimedOut(err) {
+		_, err = conn.AttachCustomerManagedPolicyReferenceToPermissionSet(input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error attaching Customer Managed Policy to SSO Permission Set (%s): %w", permissionSetArn, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s,%s,%s,%s", policyName, policyPath, permissionSetArn, instanceArn))
+
+	// Provision ALL accounts after attaching the managed policy
```

Comment:
Could you please rephrase this comment a bit to in order to make it more understandable what actually needs to happen